src/browser/api/channel.ts

```diff
@@ -134,10 +134,14 @@ export module Channel {
 
     export function connectToChannel(identity: Identity, payload: any, messageId: number, ack: AckFunc, nack: NackFunc): void {
         const { channelName, payload: connectionPayload } = payload;
-        const { uuid, name } = identity;
 
+        const connectingWindow = getExternalOrOfWindowIdentity(identity);
         const providerIdentity = Channel.getChannelByChannelName(channelName);
 
+        if (connectingWindow.isExternal && payload.customExternalWindowName) {
```

Comment:
Is there a reason we are being so specific with this property name? As we discussed, opening a channel with an adapter-specified name can serve multiple purposes. The obvious one being creating a connection per "external window"; however, this is not the only use case.

An external connection may want to create a channel per control or channel per object completely independent of windowing constructs. Suggest: `payload.customName`, `payload.nameAlias`, or something similar.

Reply:
+1

Reply:
Changing to `externalNameAlias`

Reply:
Changing to `nameAlias`